### PR TITLE
Update recipe for py312_patch

### PR DESCRIPTION
**Pull request by cbouss**
Preparation for py312 release.

### CODE REVIEW

**Review by varlackc — COMMENTED** [review]
There seem to be some redirects in the recipe also, we seem to be missing the license_family in the recipe.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,39 +1,42 @@
...
     - fastcluster
+  commands:
+    - pip check
+  requires:
+    - pip
 
 about:
-  home: http://danifold.net/fastcluster.html
+  home: https://danifold.net/fastcluster.html
   license: BSD-2-Clause AND GPL-2.0-or-later
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: I think the license is only BSD-2-Clause, see https://github.com/dmuellner/fastcluster/blob/v1.2.6/README.txt#L24 and https://github.com/dmuellner/fastcluster/blob/v1.2.6/COPYING.txt

The license was changed in v1.1.8, see https://github.com/dmuellner/fastcluster/commit/ff8d0bff6a861ef60083f6a80bdba0504f39fda9#diff-95c20d744db732cdbca24c3e0406c10005ecf7fe8b5719c2fdf2b8af3fcedc79R114

**Review by skupr-anaconda — APPROVED** [review]
LGTM but `license`

File: recipe/meta.yaml (hunk 1 above)
Comment (Jrice1317) [inline recipe/meta.yaml]:
```suggestion
  license: BSD-2-Clause
```

**Review by Jrice1317 — COMMENTED** [review]
The `doc_url` forces automatically downloads the pdf, so maybe we should use the following link instead. 
https://cran.r-project.org/web/packages/fastcluster/vignettes/fastcluster.pdf

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,39 +1,42 @@
...
   license_file: COPYING.txt
```
Comment (Jrice1317) [inline recipe/meta.yaml:42]:
```suggestion
  license_family: BSD
  license_file: COPYING.txt
```

**Review by Jrice1317 — APPROVED** [review]

**Pull request merged**